emcc.py

```diff
@@ -1564,7 +1569,7 @@ def compile_source_file(i, input_file):
               temp_files[pos] = (temp_files[pos][0], new_temp_file)
 
       # Decide what we will link
-      stop_at_bitcode = final_suffix not in EXECUTABLE_SUFFIXES
+      stop_at_bitcode = final_suffix not in EXECUTABLE_SUFFIXES and not shared.Settings.SIDE_MODULE
```

Comment:
this line looks surprising. it seems like we should add a method that checks the suffix, and call it from here, and in there have a comment about this.